qp_shogun/sortmerna/utils.py

```diff
@@ -0,0 +1,177 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+# -----------------------------------------------------------------------------
+# This file contains functions used by multiple commands
+# -----------------------------------------------------------------------------
+from qiita_client.util import system_call, get_sample_names_by_run_prefix
+from itertools import zip_longest
+from os.path import basename, join, exists
+from functools import partial
+from qiita_client import ArtifactInfo
+
+
+def make_read_pairs_per_sample(forward_seqs, reverse_seqs, map_file):
+    """Recovers read pairing information
+
+    Parameters
+    ----------
+    forward_seqs : list of strs
+        The list of forward seqs filepaths
+    reverse_seqs : list of str
+        The list of reverse seqs filepaths
+    map_file : str
+        The path to the mapping file
+
+    Returns
+    -------
+    samples: list of tup
+        list of 4-tuples with run prefix, sample name, fwd read fp, rev read fp
+
+    Raises
+    ------
+    ValueError
+        If the rev is not an empty list and the same length than fwd seqs
+        The prefixes of the run_prefix don't match the file names
+
+    Notes
+    -----
+    At this stage it is required that if reverse sequences are present that all
+    samples have both a forward and a reverse sequence. However, the read
+    trimming/filtering step can sometimes eliminate all reverse reads,
+    especially in low
+    coverage samples with poor overall reverse read quality.
+    """
+
+    # sort forward seqs
+    forward_seqs.sort()
+
+    # check that rev seqs are same len
+    if reverse_seqs:
+        if len(forward_seqs) != len(reverse_seqs):
+            raise ValueError('Your reverse and forward files are of different '
+                             'length. Forward: %s. Reverse: %s.' %
```

Comment:
The word length can be confusing in the context of sequence files, I would suggest something like:

```suggestion
            raise ValueError('There's a mistmatching count of forward and reverse files.'
                             'Forward: %s. Reverse: %s.' %
```